Source/SwiftLintFrameworkTests/TestHelpers.swift

```diff
@@ -65,6 +70,18 @@ extension XCTestCase {
 
         // Triggering examples violate
         XCTAssertEqual(triggers.flatMap({ violations($0, ruleDescription) }).count, triggers.count)
+        let triggersWithMarkers = triggers.filter { $0.containsString(violationMarker) }
+
+        // Triggering examples with violation markers violate at the marker's location
+        for trigger in triggersWithMarkers {
+            let firstViolation = violations(trigger, ruleDescription).first!
```

Comment:
Maybe we should combine this and the `triggers.flatMap` above to only calculate the violations once?